src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -60,81 +50,85 @@ export interface CheckboxInputProps {
   /**
    * Callback function when input is blurred.
    */
-  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
   /**
    * Callback function when input is focused.
    */
-  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
+  /**
+   * The size of the checkbox.
+   */
+  size?: CheckboxSize;
 }
 
-export const CheckboxInput: FC<CheckboxInputProps> = ({
+export const CheckboxInput: React.FC<CheckboxInputProps> = ({
   id,
   isChecked,
   label,
   onChange,
   className = '',
-  displayInline = false,
   error = false,
   hideLabel = false,
   helpText,
   isDisabled = false,
   isRequired = false,
   onBlur = undefined,
   onFocus = undefined,
+  size = 'md',
 }) => {
-  const handleBlur = (event: FocusEvent<HTMLInputElement>): void => {
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>): void => {
     if (onBlur) onBlur(event);
   };
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
     onChange(event);
   };
 
-  const handleFocus = (event: FocusEvent<HTMLInputElement>): void => {
+  const handleFocus = (event: React.FocusEvent<HTMLInputElement>): void => {
     if (onFocus) onFocus(event);
   };
 
-  const wrapperClasses = classNames(
-    styles.checkbox,
-    { [styles.disabled]: isDisabled },
-    { [styles.inline]: displayInline },
-  );
-
-  const inputProps = {
-    'aria-invalid': !!error,
-    'aria-label': label,
-    'aria-labelledby': label ? `${id}Label` : undefined,
+  const checkboxProps = {
     id,
-    checked: !!isChecked,
-    disabled: isDisabled,
+    isChecked: !!isChecked,
+    isDisabled,
     onBlur: handleBlur,
     onChange: handleChange,
     onFocus: handleFocus,
-    type: 'checkbox',
-    className: styles.input,
+    size,
+    label,
+    className: 'm-right-xs',
+    error,
   };
 
+  let labelMargin;
+
+  if (size === 'sm') {
+    labelMargin = '0';
+  } else if (size === 'lg') {
+    labelMargin = 'xs 0 0 0';
+  } else {
+    labelMargin = '2xs 0 0 0';
+  }
+
   const labelProps = {
     isFieldRequired: isRequired,
-    className: styles['checkbox-label'],
     inputId: id,
     hasError: !!error,
     helpText,
     isDisabled,
+    margin: labelMargin,
   };
 
   return (
     <Box className={className}>
```

Comment:
Hey @nathanyoung maybe I'm not reading this correctly, but it seems we're still not spreading props on this one. Or opening up the prop interface for `checkboxProps`.

Reply:
doh, did it for the Checkbox, but not for CheckboxInput. Updated now.

Reply:
hmm still not seeing it?

Reply:
this conversation is threaded on "outdated" code now. going to resolve it.